### ISSUE TITLE
Bounding box is shuffled. How to sort it?

### ISSUE THREAD

**Issue by kalai2033** [clovaai/CRAFT-pytorch#43]
First of all thanks very much for your model. I have used your pretrained model for text detection. I have got the bounding box coordinates as txt files. The resulted bounding boxes are shuffled and i could not sort it . When there is a presence of curved text in a same line like in the image below, the order gets shuffled and i need to sort it before passing it to text extraction model.

![image](https://user-images.githubusercontent.com/49688647/64243046-f47b8580-cf06-11e9-975b-d3e0e9afca39.png)

I have not used POLY mode. For the above image, the model outputs a txt file in which the bb coordinates are as follows. I have added the detected text for better explanation of my problem. In this case the detection order is  

146,36,354,34,354,82,146,84    "Australian"

273,78,434,151,411,201,250,129       "Collection"

146,97,250,97,250,150,146,150       "vine"

77,166,131,126,154,158,99,197      "Old"

242,215,361,241,354,273,235,248     "Valley"

140,247,224,219,234,250,150,277    "Eden"

194,298,306,296,307,324,194,325     "Shiraz"

232,406,363,402,364,421,233,426     "Vintage"

152,402,216,405,215,425,151,422    "2008"

124,470,209,480,207,500,122,490     "South"

227,481,387,472,389,494,228,503     "Australia"

222,562,312,564,311,585,222,583     "Gibson"

198,564,217,564,217,584,198,584     "by"

386,570,421,570,421,600,386,600      "750 ml"

But the expected output is Australian->old->vine->collection->Eden->Valley->shiraz->2008->vintage->south->Australia->by->GIBSON->750ml.

**Comment by nbhupendra** [comment]
> @backtime92 Thanks for your suggestion, I have found a way to sort it out.

@kalai2033 Can you share code how did you solve this. really appreciate. Thanks

**Comment by ghost** [comment]
How you sort the bounding boxes...Kindly help

**Comment by mfernaal** [comment]
> How you sort the bounding boxes...Kindly help

I found this solution, it works for me. I needed to change the tolerant_factor to 80 https://stackoverflow.com/questions/39403183/python-opencv-sorting-contours

**Comment by DhruvAwasthi** [comment]
@kalai2033 It will help us all too if you can share the code to solve this issue please.
Best,
Dhruv

**Comment by kalai2033** [comment]
@DhruvAwasthi take a look at [EasyOCR](https://github.com/JaidedAI/EasyOCR). There they are using the same CRAFT model with sorted bounding boxes.

**Comment by DhruvAwasthi** [comment]
@kalai2033 thank you so much for the help! Means a lot!

**Comment by SANJITH-KUMAR-20** [comment]
@kalai2033 where to look in the EasyOCR repo for understanding the sorting process
